SPECS/openmpi/openmpi.spec

```diff
@@ -0,0 +1,1139 @@
+mkdir -p %{buildroot}%{_datadir}/modulefiles/mpi
+# Since we're doing our own substitution here, use our own definitions.
+sed 's#@LIBDIR@#%{_libdir}/%{name}#;
+     s#@ETCDIR@#%{_sysconfdir}/%{namearch}#;
+     s#@FMODDIR@#%{_fmoddir}/%{name}#;
+     s#@INCDIR@#%{_includedir}/%{namearch}#;
+     s#@MANDIR@#%{_mandir}/%{namearch}#;
+     /@PY2SITEARCH@/d;
+     s#@PY3SITEARCH@#%{python3_sitearch}/%{name}#;
+     s#@COMPILER@#openmpi-%{_arch}%{?_cc_name_suffix}#;
+     s#@SUFFIX@#%{?_cc_name_suffix}_openmpi#' \
+     <%{SOURCE1} \
+     >%{buildroot}%{_datadir}/modulefiles/mpi/%{namearch}
+
+# make the rpm config file
+install -Dpm 644 %{SOURCE4} %{buildroot}/%{macrosdir}/macros.%{namearch}
+
+# Link the fortran module to proper location
+mkdir -p %{buildroot}%{_fmoddir}/%{name}
+for mod in %{buildroot}%{_libdir}/%{name}/lib/*.mod
+do
+  modname=$(basename $mod)
+  ln -s ../../../%{name}/lib/${modname} %{buildroot}/%{_fmoddir}/%{name}/
+done
+
+# Link the pkgconfig files into the main namespace as well
+mkdir -p %{buildroot}%{_libdir}/pkgconfig
+cd %{buildroot}%{_libdir}/pkgconfig
+ln -s ../%{name}/lib/pkgconfig/*.pc .
+cd -
+
+# Create cmake dir
+mkdir -p %{buildroot}%{_libdir}/%{name}/lib/cmake/
+
+# Remove extraneous wrapper link libraries (bug 814798)
+sed -i -e s/-ldl// -e s/-lhwloc// \
+  %{buildroot}%{_libdir}/%{name}/share/openmpi/*-wrapper-data.txt
+
+# install .pth files
+mkdir -p %{buildroot}/%{python3_sitearch}/%{name}
+install -pDm0644 %{SOURCE3} %{buildroot}/%{python3_sitearch}/openmpi.pth
+
+%check
+make check
+
+%files
+%license LICENSE
+%dir %{_libdir}/%{name}
+%dir %{_sysconfdir}/%{namearch}
+%dir %{_libdir}/%{name}/bin
+%dir %{_libdir}/%{name}/lib
+%dir %{_libdir}/%{name}/lib/openmpi
+%dir %{_mandir}/%{namearch}
+%dir %{_mandir}/%{namearch}/man*
+%config(noreplace) %{_sysconfdir}/%{namearch}/*
+%{_libdir}/%{name}/bin/mpi[er]*
+%{_libdir}/%{name}/bin/ompi*
+%{_libdir}/%{name}/bin/orte[-dr_]*
+%if %{with ucx}
+%{_libdir}/%{name}/bin/oshmem_info
+%{_libdir}/%{name}/bin/oshrun
+%{_libdir}/%{name}/bin/shmemrun
+%endif
+%{_libdir}/%{name}/lib/*.so.40*
+%{_libdir}/%{name}/lib/libmca_common_ofi.so.10*
+%{_libdir}/%{name}/lib/libmca*.so.41*
+%{_libdir}/%{name}/lib/libmca*.so.50*
+%if 0%{?el7}
+%{_libdir}/%{name}/lib/pmix/
+%endif
+%{_mandir}/%{namearch}/man1/mpi[er]*
+%{_mandir}/%{namearch}/man1/ompi*
+%{_mandir}/%{namearch}/man1/orte[-dr_]*
+%if %{with ucx}
+%{_mandir}/%{namearch}/man1/oshmem_info*
+%{_mandir}/%{namearch}/man1/oshrun*
+%{_mandir}/%{namearch}/man1/shmemrun*
+%endif
+%{_mandir}/%{namearch}/man7/ompi_*
+%{_mandir}/%{namearch}/man7/opal_*
+%{_mandir}/%{namearch}/man7/orte*
+%{_libdir}/%{name}/lib/openmpi/*
+%{_datadir}/modulefiles/mpi/
+%dir %{_libdir}/%{name}/share
+%dir %{_libdir}/%{name}/share/openmpi
+%{_libdir}/%{name}/share/openmpi/amca-param-sets
+%{_libdir}/%{name}/share/openmpi/help*.txt
+%if %{with rdma}
+%{_libdir}/%{name}/share/openmpi/mca-btl-openib-device-params.ini
+%endif
+%if 0%{?el7}
```

Comment:
Let's remove all sections for `e17` - I believe it's a distro-specific macro, which will never be true for Mariner.

Reply:
updated for openmpi.spec 
Looks like they were using internal pmix instead of external for el7. fixed.